pkgs/development/libraries/mesa/default.nix

```diff
@@ -175,6 +181,17 @@ self = stdenv.mkDerivation {
     for js in $drivers/share/vulkan/icd.d/*.json; do
       substituteInPlace "$js" --replace "$out" "$drivers"
     done
+
+    # Move OpenCL stuff
+    mkdir -p $opencl/lib
+    mv -t "$opencl/lib/"     \
+      $out/lib/gallium-pipe   \
+      $out/lib/libMesaOpenCL*
+
+    # We construct our own .icd file that contains an absolute path.
+    rm -r $out/etc/OpenCL
+    mkdir -p $opencl/etc/OpenCL/vendors/
+    echo $opencl/lib/libMesaOpenCL.so > $opencl/etc/OpenCL/vendors/mesa.icd
```

Comment:
Did you test if this works? I guess this gets linked as `/run/opengl-driver/etc/OpenCL/vendors/mesa.icd` but I'm not sure if that path is searched and I haven't tested it yet.

Reply:
Yes, this worked when I tested it. The [NixOS manual](https://nixos.org/manual/nixos/stable/index.html#sec-gpu-accel-opencl) mentions `hardware.opengl.extraPackages` (which get linked to `/run/opengl-driver`) as a way for installing OpenCL ICDs, so that seems to be the intended path for them.